pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotTableTenantConfigs.java

```diff
@@ -46,15 +46,20 @@
   @Produces(MediaType.APPLICATION_JSON)
   @Path("/tables/{tableName}/rebuildBrokerResourceFromHelixTags")
   @ApiOperation(value = "Rebuild broker resource for table", notes = "when new brokers are added")
-  @ApiResponses(value = {@ApiResponse(code=200, message = "Success"),
+  @ApiResponses(value = {
+      @ApiResponse(code=200, message = "Success"),
+      @ApiResponse(code = 400, message = "Broker resource cannot be rebuilt because ideal state is the same for this table"),
       @ApiResponse(code = 404, message = "Table not found"),
       @ApiResponse(code = 500, message = "Internal error rebuilding broker resource or serializing response")})
-  public String rebuildBrokerResource(
+  public SuccessResponse rebuildBrokerResource(
       @ApiParam(value = "Table name (with type)", required = true) @PathParam("tableName") String tableNameWithType
-  ) throws JSONException {
-    final PinotResourceManagerResponse pinotResourceManagerResponse =
-        _helixResourceManager.rebuildBrokerResourceFromHelixTags(tableNameWithType);
-
-    return pinotResourceManagerResponse.toJSON().toString();
+  ) {
+    try {
+      final PinotResourceManagerResponse pinotResourceManagerResponse =
+          _helixResourceManager.rebuildBrokerResourceFromHelixTags(tableNameWithType);
+      return new SuccessResponse(pinotResourceManagerResponse.getMessage());
+    } catch (WebApplicationException e) {
```

Comment:
You should catch the exceptions that are being thrown in PinotHelixResourceManager for this call sequence. In this case, it should be InvalidConfig, in which case you return 4xx, and any other in which case you return 5xx